insights/v1alpha2/types_insights.go

```diff
@@ -0,0 +1,427 @@
+	Name string `json:"name"`
+	// lastGatherSeconds is required field that represents the time spent gathering in seconds
+	// +kubebuilder:validation:Minimum=0
+	// +required
+	LastGatherSeconds int32 `json:"lastGatherSeconds"`
+}
+
+// insightsReport provides Insights health check report based on the most
+// recently sent Insights data.
+type InsightsReport struct {
+	// downloadedTime is an optional time when the last Insights report was downloaded.
+	// An empty value means that there has not been any Insights report downloaded yet and
+	// it usually appears in disconnected clusters (or clusters when the Insights data gathering is disabled).
+	// +optional
+	DownloadedTime *metav1.Time `json:"downloadedTime,omitempty"`
+	// healthChecks provides basic information about active Insights health checks
+	// in a cluster.
+	// +listType=map
+	// +listMapKey=advisorURI
+	// +listMapKey=totalRisk
+	// +listMapKey=description
+	// +kubebuilder:validation:MaxItems=100
+	// +optional
+	HealthChecks []HealthCheck `json:"healthChecks,omitempty"`
+	// uri is optional field that provides the URL link from which the report was downloaded.
+	// The link must be a valid HTTPS URL and the maximum length is 2048 characters.
+	// +kubebuilder:validation:XValidation:rule=`isURL(self) && url(self).getScheme() == "https"`,message=`URI must be a valid HTTPS URL (e.g., https://example.com)`
+	// +kubebuilder:validation:MaxLength=2048
+	// +optional
+	URI string `json:"uri,omitempty"`
+}
+
+// healthCheck represents an Insights health check attributes.
+type HealthCheck struct {
+	// description is required field that provides basic description of the healtcheck.
+	// It must contain at least 10 characters and may not exceed 2048 characters.
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:MinLength=10
+	// +required
+	Description string `json:"description"`
+	// totalRisk is the required field of the healthcheck.
+	// It is indicator of the total risk posed by the detected issue; combination of impact and likelihood.
+	// Allowed values are Low, Medium, Important and Critical.
+	// The value represents the severity of the issue.
+	// +required
+	TotalRisk TotalRisk `json:"totalRisk"`
+	// advisorURI is required field that provides the URL link to the Insights Advisor.
+	// The link must be a valid HTTPS URL and the maximum length is 2048 characters.
+	// +kubebuilder:validation:XValidation:rule=`isURL(self) && url(self).getScheme() == "https"`,message=`advisorURI must be a valid HTTPS URL (e.g., https://example.com)`
+	// +kubebuilder:validation:MaxLength=2048
+	// +required
+	AdvisorURI string `json:"advisorURI"`
+}
+
+// totalRisk defines the valid totalRisk values.
+// +kubebuilder:validation:Enum=Low;Moderate;Important;Critical
+type TotalRisk string
+
+const (
+	TotalRiskLow       TotalRisk = "Low"
+	TotalRiskModerate  TotalRisk = "Moderate"
+	TotalRiskImportant TotalRisk = "Important"
+	TotalRiskCritical  TotalRisk = "Critical"
+)
```

Comment:
Nit: Why Low/Moderate/Important/Critical? I was expecting High to follow Moderate

Reply:
I don't have a strong opinion, but I'll ask @tremes whether this naming was introduced in IO and if it's okay to change it."

Reply:
This reflects the severity levels defined for the Insights recommendation so it's not easy to change.

Reply:
Ack, that's ok then, just felt like it didn't fit, but if it's already shipped it's already shipped